### PR TITLE
Few fixes

### PR DESCRIPTION
**Pull request by ImMorpheus**
Few fixes (wip)

### CODE REVIEW

File: src/html/downloads.html [hunk 1]
```diff
@@ -39,7 +39,7 @@
     <header>
         <div class="container">
             <div class="logo">
-                <img src="assets/img/icons/spongie-mark-reverse-dark.svg" />
+                <img src="assets/img/icons/spongie-mark-reverse-dark.svg" alt="Spongie reverse"/>
```
Comment (jamierocks) [inline src/html/downloads.html]: This really isn't required, not going to help accessibility at all.

File: src/html/include/base.html [hunk 2]
```diff
@@ -55,7 +55,7 @@
     <div class="container">
         <div id="sp-logo-container" class="page-scroll">
             <a class="logo" href="#top">
-                <img src="assets/img/icons/spongie-mark.svg" style="height: 40px">
+                <img src="assets/img/icons/spongie-mark.svg" alt="Spongie" style="height: 40px">
```
Comment (jamierocks) [inline src/html/include/base.html]: This really isn't required, not going to help accessibility at all.

File: src/js/downloads/Downloads.vue [hunk 3]
```diff
@@ -30,7 +30,7 @@
             <div class="row">
                 <div class="col-lg-5 col-md-6">
                     <div class="logo">
-                        <img src="assets/img/icons/spongie-mark-reverse-dark.svg" />
+                        <img src="assets/img/icons/spongie-mark-reverse-dark.svg" alt="Spongie reverse" />
```
Comment (jamierocks) [inline src/js/downloads/Downloads.vue]: This really isn't required, not going to help accessibility at all.

File: src/js/downloads/Main.vue [hunk 4]
```diff
@@ -30,7 +30,7 @@
             <div class="row">
                 <div class="col-lg-5 col-md-6">
                     <div class="logo">
-                        <img src="assets/img/icons/spongie-mark-reverse-dark.svg" />
+                        <img src="assets/img/icons/spongie-mark-reverse-dark.svg" alt="Spongie reverse"/>
```
Comment (jamierocks) [inline src/js/downloads/Main.vue]: This really isn't required, not going to help accessibility at all.

**Review by jamierocks — COMMENTED** [review]
This should really be commented on our OCD issue.

**Comment by ImMorpheus** [comment]
The alt attribute is required for img tag. However if you really don't like the content we can use empty alt.

Edit: I forgot about the ocd list, thanks for that

**Comment by jamierocks** [comment]
Yes they should have an empty alt

**Comment by jamierocks** [comment]
As of HTML5 self-closed tags (such as img) no longer need to use specific syntax (`/>`), and can just be left as a single tag.

**Comment by ImMorpheus** [comment]
Note: currently I had to add in the CSP connect-src for the download page and frame-src for kiwilrc. Everything works, if we want to make it more restrictive we can move the fonts on the server and use font-src 'self'

**Comment by jamierocks** [comment]
> if we want to make it more restrictive we can move the fonts on the server and use font-src 'self'

That'd be overkill

**Comment by jamierocks** [comment]
Not to demean your pr, but these changes are the type of things that an OCD issue exists for in every SpongePowered repository ([SpongeHome issue](https://github.com/SpongePowered/SpongeHome/issues/104)) for. 

Perhaps exception can be made of this pr, in light of the number of issues resolved - however in future that is certainly where these issues should be documented.

**Comment by ImMorpheus** [comment]
Yeah, you've already said that (and I added the issues to the OCD list), but if you really want to fix them I can delete the pull request and no 'exception' will be made. 
That being said, I fixed a few things and created this pull because I knew changes would be over 20 lines. [Thank goodness small pull request get closed and people informed they should use the OCD.](https://github.com/SpongePowered/SpongeHome/pull/122)

**Comment by jamierocks** [comment]
> Thank goodness small pull request get closed and people informed they should use the OCD.

In regards to that pr, I am not sure why that was merged and not directed to the OCD issue. It should have been.

File: controllers/header.go [hunk 5]
```diff
@@ -26,20 +26,17 @@
...
 
 func AddHeaders(resp http.ResponseWriter) {
 	header := resp.Header()
 
-	// TODO: Needs more testing and a few changes to make it more restrictive
-	/*header.Add("Content-Security-Policy", "default-src 'self' https:; "+
-	"style-src 'self' 'unsafe-inline' https:; "+
-	"script-src 'self' 'unsafe-eval' https://cdnjs.cloudflare.com https://www.google-analytics.com; "+
-	"frame-ancestors 'none'")*/
+	header.Set("Content-Security-Policy", "default-src 'none'; script-src 'self' 'unsafe-inline' 'unsafe-eval' https://cdnjs.cloudflare.com/ ; style-src 'self' 'unsafe-inline' https://cdnjs.cloudflare.com; img-src 'self'; font-src https://cdnjs.cloudflare.com/ https://fonts.googleapis.com/; connect-src 'self' https://dl-api.spongepowered.org/; frame-src https://kiwiirc.com; frame-ancestors 'none'; upgrade-insecure-requests; block-all-mixed-content; ")
```
Comment (jamierocks) [inline controllers/header.go]: Can we get this split over a few lines, to promote readability?

File: src/html/downloads.html [hunk 6]
```diff
@@ -39,7 +39,7 @@
     <header>
         <div class="container">
             <div class="logo">
-                <img src="assets/img/icons/spongie-mark-reverse-dark.svg" />
+                <img src="assets/img/icons/spongie-mark-reverse-dark.svg" alt=""/>
```
Comment (jamierocks) [inline src/html/downloads.html]: Change `/>` for `>` as per HTML5.

File: src/html/include/base.html [hunk 7]
```diff
@@ -129,7 +129,7 @@
                 <p>Copyright &copy; SpongePowered 2014-2017</p>
                 <p id="fastly">Accelerated by<br>
                     <a href="https://www.fastly.com">
-                        <img src="assets/img/fastly.png" alt="Fastly" />
+                        <img src="assets/img/fastly.png" alt="Fastly" height="45" width="100" />
```
Comment (jamierocks) [inline src/html/include/base.html]: Change `/>` for `>` as per HTML5.

File: src/js/downloads/Downloads.vue [hunk 8]
```diff
@@ -30,7 +30,7 @@
             <div class="row">
                 <div class="col-lg-5 col-md-6">
                     <div class="logo">
-                        <img src="assets/img/icons/spongie-mark-reverse-dark.svg" />
+                        <img src="assets/img/icons/spongie-mark-reverse-dark.svg" alt="" />
```
Comment (jamierocks) [inline src/js/downloads/Downloads.vue]: Change `/>` for `>` as per HTML5.

File: src/js/downloads/Main.vue [hunk 9]
```diff
@@ -30,7 +30,7 @@
             <div class="row">
                 <div class="col-lg-5 col-md-6">
                     <div class="logo">
-                        <img src="assets/img/icons/spongie-mark-reverse-dark.svg" />
+                        <img src="assets/img/icons/spongie-mark-reverse-dark.svg" alt=""/>
```
Comment (jamierocks) [inline src/js/downloads/Main.vue]: Change `/>` for `>` as per HTML5.

File: README.md [hunk 10]
```diff
@@ -1,20 +1,19 @@
...
 ==========
 
-The Sponge Project [website](https://www-staging.spongepowered.org/), licensed
+The Sponge Project [website](https://www.spongepowered.org/), licensed
 under the MIT license. See [LICENSE.md](LICENSE.md) for details.
 
 ### Branches
 
-- `master` is hooked up to [www staging](https://www-staging.spongepowered.org/)
 - `production` is hooked up to [www prod](https://www.spongepowered.org/)
```
Comment (jamierocks) [inline README.md]: The production branch is no longer used - master is now `www prod`.

**Review by jamierocks — CHANGES_REQUESTED** [review]

File: .gitignore [hunk 11]
```diff
@@ -80,13 +80,13 @@ lib-cov
 # Coverage directory used by tools like istanbul
 coverage
 
-# Grunt intermediate storage (http://gruntjs.com/creating-plugins#storing-task-files)
+# Grunt intermediate storage (https://gruntjs.com/creating-plugins#storing-task-files)
```
Comment (jamierocks) [inline .gitignore]: May as well remove Grunt from the gitignore - we don't use it.

**Review by jamierocks — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 11 above)
Comment (ImMorpheus) [reply .gitignore]: Is that okay ? You created that gitignore with gitignore.io, it is basically a preset which handle multiple developers using different ide/tools/etc

Comment (jamierocks) [reply .gitignore]: Yeah - Grunt is a buildtool, we use Gulp so Grunt will never become part of our dev env.

Comment (ImMorpheus) [reply .gitignore]: Done

Comment (ImMorpheus) [inline .gitignore]: Shall I open an issue if I was able to return the nginx version in the home or shall I contact you privately ? (Yeah I know it's off topic, I didn't know where to ask this)

File: .gitignore [hunk 12]
```diff
@@ -80,13 +80,10 @@ lib-cov
 # Coverage directory used by tools like istanbul
 coverage
 
-# Grunt intermediate storage (http://gruntjs.com/creating-plugins#storing-task-files)
```
Comment (stephan-gh) [inline .gitignore]: There is no reason to remove this. The gitignore is based on a template and since this isn't breaking anything in our project I see no reason to remove it.

File: .gitignore [hunk 13]
```diff
@@ -80,13 +80,10 @@ lib-cov
...
-.grunt
-
 # node-waf configuration
 .lock-wscript
 
-# Compiled binary addons (http://nodejs.org/api/addons.html)
+# Compiled binary addons (https://nodejs.org/api/addons.html)
```
Comment (stephan-gh) [inline .gitignore]: Feel free to suggest this change in the upstream repository: https://github.com/github/gitignore/blob/master/Node.gitignore#L32. I don't see a reason to change it here.

File: controllers/header.go [hunk 14]
```diff
@@ -26,20 +26,23 @@
...
 
 func AddHeaders(resp http.ResponseWriter) {
 	header := resp.Header()
 
-	// TODO: Needs more testing and a few changes to make it more restrictive
-	/*header.Add("Content-Security-Policy", "default-src 'self' https:; "+
-	"style-src 'self' 'unsafe-inline' https:; "+
-	"script-src 'self' 'unsafe-eval' https://cdnjs.cloudflare.com https://www.google-analytics.com; "+
-	"frame-ancestors 'none'")*/
+	header.Set("Content-Security-Policy", "default-src 'none'; " +
```
Comment (stephan-gh) [inline controllers/header.go]: Please move this to a separate PR, it needs careful testing and can't be accepted together with the other minor changes.

File: .gitignore (hunk 12 above)
Comment (ImMorpheus) [reply .gitignore]: I knew it would end up this way

File: controllers/header.go (hunk 14 above)
Comment (ImMorpheus) [reply controllers/header.go]: Oh ok, I didn't open a new pull because it was a rather small fix (in terms of lines)

File: controllers/header.go [hunk 15]
```diff
@@ -49,3 +49,54 @@ func AddHeaders(resp http.ResponseWriter) {
 		header.Add(fastly.SurrogateControlHeader, cache.SurrogateStaticContentOptions)
 	}
 }
+/*
```
Comment (stephan-gh) [inline controllers/header.go]: I think this is not what you wanted to do :)

Comment (ImMorpheus) [reply controllers/header.go]: Yeah, I noticed that soon after

**Comment by kashike** [comment]
You should not be changing any of the sponsor images. :no_good_man:

**Comment by ImMorpheus** [comment]
I just compressed the images, can't I do that ? (Not a sensitive change, but now they are compressed)
Also, is the backgroud included in the group of images I shouldn't change ?

**Comment by stephan-gh** [comment]
@ImMorpheus It's probably better if we keep them as-is, because they were not made by us. Regarding the background, it seems like your optimized image has the same size as the original one?

**Comment by ImMorpheus** [comment]
Ok, I didn't know that. Consider it fixed
The backgroud image is 1% smaller, we are talking about nearly 500B (that's why I thought it was not a big deal)

**Comment by stephan-gh** [comment]
@ImMorpheus 500 bytes isn't really worth changing the image, that is almost nothing.

File: src/html/include/base.html [hunk 16]
```diff
@@ -129,7 +129,7 @@
                 <p>Copyright &copy; SpongePowered 2014-2017</p>
                 <p id="fastly">Accelerated by<br>
                     <a href="https://www.fastly.com">
-                        <img src="assets/img/fastly.png" alt="Fastly" />
+                        <img src="assets/img/fastly.png" alt="Fastly" height="45" width="100">
```
Comment (stephan-gh) [inline src/html/include/base.html:132]: Why did you add an explicit width/height?

**Comment by ImMorpheus** [comment]
Yeah, I compressed the images all together and there was a compression of more than 20% with some of the others. It just happened to be there and was compressed

File: src/html/include/base.html (hunk 16 above)
Comment (ImMorpheus) [reply src/html/include/base.html:132]: If height and width are set, the space required for the image is reserved when the page is loaded. Thus, the page loads faster.

**Comment by stephan-gh** [comment]
@ImMorpheus That's fine but please revert the change. We shouldn't require all clients to download the whole image again just for 500 bytes.

**Comment by ImMorpheus** [comment]
Done

**Review by stephan-gh — APPROVED** [review]
